ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/commitmarkable/integration/CMRIntegrationTest.java

```diff
@@ -169,60 +167,40 @@ public void run() {
 									.createStatement();
 							createStatement
 									.execute("INSERT INTO foo (bar) VALUES (1)");
-							// System.out.printf("XXX txn close%n");
-
-							if (faultType == 1)
-								Runtime.getRuntime().halt(0);
 
                             userTransaction.commit();
 							connection.close(); // This wouldn't work for a
 												// none-JCA code as commit has
 												// closed the connection - it
 												// helps us though as JCA seems
 												// to rely on finalize
-
-							// System.out
-							// .printf("committed txn iteration %d%n", i);
 							success++;
 						} catch (SQLException e) {
-							System.err.println("boom");
-							e.printStackTrace();
-							if (e.getCause() != null) {
-								e.getCause().printStackTrace();
-							}
+							log.errorf(e, "Error while invoking insertion to a database table with CMR resource");
 							SQLException nextException = e.getNextException();
 							while (nextException != null) {
-								nextException.printStackTrace();
-								nextException = nextException
-										.getNextException();
-							}
-							Throwable[] suppressed = e.getSuppressed();
-							for (int j = 0; j < suppressed.length; j++) {
-								suppressed[j].printStackTrace();
+								log.errorf("Next SQLException chained", nextException);
+								nextException = nextException.getNextException();
 							}
 							try {
 								userTransaction.rollback();
 							} catch (IllegalStateException | SecurityException
 									| SystemException e1) {
-								e1.printStackTrace();
-								fail("Problem with transaction");
+								log.error("Problem with transaction", e1);
+								throw new IllegalStateException("Problem with transaction", e1);
```

Comment:
I would prefer that we explicitly fail here (also there are other places where we throw instead of fail). If a test throws an exception I interpret that as meaning that the test needs fixing unless the test is marked as such, eg @Test (expected = IllegalStateException.class).

Reply:
Sure. Fixed.